src/hotspot/share/opto/parse1.cpp

```diff
@@ -737,7 +737,7 @@ void Parse::do_all_blocks() {
 
         if (DoPartialEscapeAnalysis && block->is_loop_head()) {
           PEAState& as = jvms()->alloc_state();
-          as.mark_all_escaped();
+          as.mark_all_live_objects_escaped(PEA(), block->start_map());
```

Comment:
should we call kill_dead_locals() before this? 
It will kill dead locals based on liveness information.

Reply:
Sure. Seems logical to me.